subprojects/docs/src/docs/userguide/running-builds/build_environment.adoc

```diff
@@ -73,6 +73,18 @@ _Default is `10800000` (3 hours)._
 When set to `true`, Gradle will run the build with remote debugging enabled, listening on port 5005. Note that this is the equivalent of adding `-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005` to the JVM command line and will suspend the virtual machine until a debugger is attached.
 _Default is `false`._
 
+`org.gradle.debug.port=(port number)`::
+Specifies the port number on which process is listening when debug is enabled.
+_Default is `5005`._
+
+`org.gradle.debug.server=(true,false)`::
+Then set to `true` and debugging enabled, Gradle will run the build with socked-attach mode of debugger. Otherwise socket-listen is used.
```

Comment:
```suggestion
If set to `true` and debugging is enabled, Gradle will run the build with the socket-attach mode of the debugger. Otherwise, the socket-listen mode is used.
```